contracts/test/ocr2/aggregator.test.ts

```diff
@@ -237,19 +243,62 @@ describe('aggregator.cairo', function () {
     })
   })
 
+  it("should emit 'new_transmission' event on transmit", async () => {
+    const txHash = await transmit(1, 99)
+    const receipt = await starknet.getTransactionReceipt(txHash)
+
+    assert.isNotEmpty(receipt.events)
+    console.log("Log raw 'new_transmission' event: ")
+    console.log(receipt.events[0])
+
+    const decodedEvents = await aggregator.decodeEvents(receipt.events)
+    assert.isNotEmpty(decodedEvents)
+    console.log("Log decoded 'new_transmission' event: ")
+    console.log(decodedEvents[0])
+
+    const e = decodedEvents[0]
+    const transmitter = oracles[0].transmitter.address
+
+    // NOTICE: Leading zeros are trimmed for an encoded felt (number).
+    //   To decode, the raw felt needs to be start padded up to max felt size (252 bits or < 32 bytes).
+    const hexPadStart = (data: any, len: number) => {
+      return `0x${data.toString(16).padStart(len, '0')}`
+    }
+
+    assert.equal(e.name, 'new_transmission')
+    assert.equal(e.data.round_id, 1n)
+
+    let len = 32 * 2 // transmitter
+    assert.equal(hexPadStart(e.data.transmitter, len), transmitter)
+    assert.equal(e.data.observation_timestamp, 1n)
+
+    len = 31 * 2 // observers
+    assert.equal(
+      hexPadStart(e.data.observers, len),
+      '0x00010203000000000000000000000000000000000000000000000000000000',
+    )
```

Comment:
I think to decode this properly:

```js
e.data.observers.toArray('be', 32).slice(0, e.data.observations_len)
```

So `felt` -> `byte[32]` -> `byte[observations_len]`